utils/v2/connector/impl/fetchers/StreamrFetcher.ts

```diff
@@ -0,0 +1,65 @@
+import StreamrClient from "streamr-client";
+import _ from "lodash";
+import { Fetcher, SignedDataPackageResponse, SourceConfig } from "./Fetcher";
+
+export class StreamrFetcher extends Fetcher {
+  protected lastValue?: SignedDataPackageResponse;
+  protected streamrClient: StreamrClient;
+
+  constructor(config: SourceConfig, asset?: string) {
+    super(config, asset);
+    this.streamrClient = new StreamrClient({
+      auth: {
+        privateKey: StreamrClient.generateEthereumAccount().privateKey,
+      },
+    });
+  }
+
+  init() {
+    const streamId = this.getStreamId();
+    this.streamrClient.subscribe(
+      streamId,
+      (value: any) => {
+        console.log(`Received new value from: ${streamId}`);
+        this.lastValue = this.extractPriceValue(value);
+      });
+    console.log(`Subscribed to streamr: ${streamId}`);
+  }
+
+  async getLatestData(): Promise<SignedDataPackageResponse> {
+    if (!this.lastValue) {
+      throw new Error("No data received from stream yet");
+    }
+    return this.lastValue;
+  }
+
+  protected getStreamId() {
+    return `${this.config.streamrEndpointPrefix!}/`
+      + (this.asset ? "prices" : "package");
+  }
+
+  protected extractPriceValue(receivedValue: any): SignedDataPackageResponse {
```

Comment:
Do we only stream single prices?

Reply:
No. Why do you think so?

Reply:
My mistake, I haven't noticed the else part